### ISSUE TITLE
Kitchen converge fails on Windows Box

### ISSUE THREAD

**Issue by ankitgupta07** [test-kitchen/test-kitchen#1056]
While testing out a windows based cookbook locally, I am running into the below error. This is happening only for one specific cookbook but rest of the Windows based cookbooks work fine. 

## Kitchen Version

kitchen version - 1.8.0

# kitchen converge output
```ruby
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #converge action: [No such file or directory @ rb_sysopen - C:/Users/agupta/AppData/Local/Temp/default-windows-2008R2-sandbox-20160615-6260-jo76bq/cookbooks/windows-jenkins-slave-dotnet/.kitchen/kitchen-vagrant/kitchen-windows-jenkins-slave-dotnet-default-windows-2008R2/.vagrant/machines/default/virtualbox/action_set_name]
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

```

## ChefDK Version
```ruby
Chef Development Kit Version: 0.12.0
```
## Platform Version

I am running test kitchen on Windows 7.

## Replication Case
I am running a test kitchen for home developed cookbook. I would not be able to share the cookbook details but if you want I can share my kitchen.yml file.

## Kitchen debug Output
```ruby
$ kitchen converge -l debug
-----> Starting Kitchen (v1.8.0)
D      [Vagrant command] BEGIN (vagrant --version)
D      [Vagrant command] END (0m0.00s)
D      [Vagrant command] BEGIN (vagrant plugin list)
D      [Vagrant command] END (0m0.00s)
D      Berksfile found at c:/chef-repo/cookbooks/windows-jenkins-slave-dotnet/Be
rksfile, loading Berkshelf
D      Berkshelf 4.3.5 library loaded
D      winrm requested, loading winrm gem (["~> 1.6"])
D      winrm is loaded.
D      winrm-fs requested, loading winrm-fs gem (["~> 0.4.1"])
D      winrm-fs is loaded.
-----> Converging <default-windows-2008R2>...
       Preparing files for transfer
D      Creating local sandbox in C:/Users/agupta/AppData/Local/Temp/default-wind
ows-2008R2-sandbox-20160615-4236-159q65x
       Preparing dna.json
D      Creating dna.json from {:run_list=>["recipe[windows-jenkins-slave-dotnet:
:default]"]}
       Resolving cookbook dependencies with Berkshelf 4.3.5...
D      Using Berksfile from c:/chef-repo/cookbooks/windows-jenkins-slave-dotnet/
Berksfile
D      Cleaning up local sandbox in C:/Users/agupta/AppData/Local/Temp/default-w
indows-2008R2-sandbox-20160615-4236-159q65x
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #converge action: [No such file or directory
@ dir_s_mkdir - C:/Users/agupta/AppData/Local/Temp/default-windows-2008R2-sandbo
x-20160615-4236-159q65x/cookbooks/windows-jenkins-slave-dotnet/.kitchen/kitchen-
vagrant/kitchen-windows-jenkins-slave-dotnet-default-windows-2008R2/.vagrant/mac
hines/default/virtualbox]
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

D      ------Exception-------
D      Class: Kitchen::ActionFailed
D      Message: Failed to complete #converge action: [No such file or directory
@ dir_s_mkdir - C:/Users/agupta/AppData/Local/Temp/default-windows-2008R2-sandbo
x-20160615-4236-159q65x/cookbooks/windows-jenkins-slave-dotnet/.kitchen/kitchen-
vagrant/kitchen-windows-jenkins-slave-dotnet-default-windows-2008R2/.vagrant/mac
hines/default/virtualbox]
D      ---Nested Exception---
D      Class: Errno::ENOENT
D      Message: No such file or directory @ dir_s_mkdir - C:/Users/agupta/AppDat
a/Local/Temp/default-windows-2008R2-sandbox-20160615-4236-159q65x/cookbooks/wind
ows-jenkins-slave-dotnet/.kitchen/kitchen-vagrant/kitchen-windows-jenkins-slave-
dotnet-default-windows-2008R2/.vagrant/machines/default/virtualbox
D      ------Backtrace-------
```

## Kitchen Diagnose

Please include the output of `kitchen diagnose --all` or `kitchen diagnose --loader` (if the first failed).  This can be helpful in troubleshooting, so please include it in a gist.

[gem.txt](https://github.com/test-kitchen/test-kitchen/files/315928/gem.txt)
[diagnose.txt](https://github.com/test-kitchen/test-kitchen/files/315929/diagnose.txt)

**Comment by smurawski** [comment]
@ankitgupta07 Do you have a .chefignore file in your cookbook that excludes the .kitchen folder?

It looks like kitchen is trying to copy over the .kitchen dir as well.

**Comment by ankitgupta07** [comment]
I do have a chefignore file in my cookbook. The file didn't include .kitchen folder. I tried adding below to the chefignore file but that didn't help.

```
.kitchen/*
```

**Comment by smurawski** [comment]
@ankitgupta07 Try `.kitchen*` in your chefignore file.  That's what's dropped in by the `chef generate cookbook` command and seems to work for me.

**Comment by ankitgupta07** [comment]
Thanks. That fixed my problem but may I know what was the issue because as far as I know I don't have `.kitchen*` in any of my windows based cookbooks and they work perfectly fine. This seem to be a rare occurrence. Would be good to know the reason.

**Comment by smurawski** [comment]
Without that in the chefignore, Berkshelf includes the .kitchen folder as part of the cookbook and copies it over to your sandbox before copying it to the system under test..  

Outside of just making your converge faster (since you don't have to copy a vm image locally and over the wire), trying to copy vm files that are in use is potentially problematic depending on the type of access happening the vm files at the time.  

Basically, the failure or lack thereof depend on what is happening in the virtualization application which kitchen really doesn't have control over.

Glad you got it working!
